### ISSUE TITLE
One of version 0.6.5 does not work

### ISSUE THREAD

**Issue by dmyger** [koreader/kpvbooklet#29]
First I took kpvbooklet v0.6.5  from https://github.com/koreader/kpvbooklet/releases
With this version it works like without any kpvbooklet installed. Tap on any *.fb2, *.pdf or *.djvu files it shows the message: **can't open file**. But this version adds a KUAL menu extension.
Then I found another 0.6.5 version on https://www.mobileread.com/forums/showthread.php?t=285262 it works fine, like it was with 0.6.4, but has **no** the KUAL submenu.
So, you have two releases with same version, but with different behaviour.

Kindle Whitepaper 3 (5.6.5) and Koreader's version from 04/30/2017: koreader-kindle-arm-linux-gnueabi-v2015.11-1023-g773e24f.zip

**Comment by woodpenker** [comment]
I have the same problem when open pdf files with koreader. With 0.6.5 version from https://github.com/koreader/kpvbooklet/releases, it shows the same message.So I reinstall the 0.6.4 version, it can open the pdf filles,but I can't find the KUAL submenu too.
My kindle is KPW2 (version 5.7.2.1)
My KUAL version is v2.7(20160721)
My koreader is koreader-kindle-arm-linux-gnueabi-v2015.11-1023-g773e24f

**Comment by cytown** [comment]
Could you please take a try for this:

[kindle-kpvbooklet-0.6.5.zip](https://github.com/koreader/kpvbooklet/files/1029131/kindle-kpvbooklet-0.6.5.zip)

**Comment by dmyger** [comment]
This one is number two from my first comment. Works, but no KUAL extension.

**Comment by cytown** [comment]
That's strange, please extract this file and put in extensions....

[extensions.zip](https://github.com/koreader/kpvbooklet/files/1029158/extensions.zip)

**Comment by woodpenker** [comment]
I just test the file you gave. It shows the KUAL extension. But when open pdf files,it still show the can't open file msg.

**Comment by cytown** [comment]
Please try use the extension to set pdf file first.

**Comment by cytown** [comment]
Yes, you can use that one.

**Comment by woodpenker** [comment]
I diff the file in my kindle's 'extension/kpvbooklet' and the code below dir 'extension/kpvbooklet'. shows nothing diff.And my situation is diff to dmyger 's.

**Comment by cytown** [comment]
Please try use the extension to set pdf file first.

**Comment by woodpenker** [comment]
I am now use 0.6.4 version with extension dir from 0.6.5 .It works, open 
PDF is ok with koreader, KUAL shows the submenu too, but still can't switch the setting. Nerver mind,I don't need switch. Maybe it is the problem of my kindle system version,which is not the latest.

**Comment by cytown** [comment]
@woodpenker Please try use 0.6.5, and set the switch settings first, and give the feedback.

**Comment by woodpenker** [comment]
@cytown I tried it,still failed when switch settings first.But I can open the pdf using koreader when enter the submenu: koreader_setting-koreader_file_explorer.

**Comment by cytown** [comment]
@woodpenker thank you, I will check...

**Comment by cytown** [comment]
Please check whether these files exists:

/mnt/us/koreader/koreader.sh
/mnt/us/extensions/kpvbooklet/bin/booklet.ini
/var/local/mkk/gandalf
/var/local/mkk/su

And provide the contents of booklet.ini.

**Comment by cytown** [comment]
Also, please check and provide the last 50 lines of file /tmp/kpvbooklet.log.

**Comment by woodpenker** [comment]
@cytown 
```
/mnt/us/koreader/koreader.shb --no exists
/mnt/us/extensions/kpvbooklet/bin/booklet.ini --no exists
/var/local/mkk/gandalf --no exists
/var/local/mkk/su --no exists
```

**Comment by cytown** [comment]
/mnt/us/koreader/koreader.sh don't exist???
Have you installed koreader?
Also, su not install???

**Comment by woodpenker** [comment]
@cytown I am using koreader,I'm sure. I don't know why I don't have them.I also cracked my kindle.But I can't find the file u said.

**Comment by cytown** [comment]
@chrox Please check this, I only have kpw2, so is there something different in koreader and su??

**Comment by dmyger** [comment]
On my Kindle those files exists:
_/mnt/us/koreader/koreader.sh
/var/local/mkk/gandalf
/var/local/mkk/su_
But no:
_/mnt/us/extensions/kpvbooklet/bin/booklet.ini_
more over there is no any **bin** folder inside **kpvbooklet** extension. Tried with kpvbooklet 6.5 from github releases.

**Comment by Summerslyb** [comment]
@cytown It seems that You forgot to put bin/ into extensions/kpvbooklet.
In extensions/kpvbooklet, Only menu.json and config.xml exist there.
After checking 85e4940a8a59fa4c4e593f8524736808f150751d and dcb9bf280e4de8c12a93479f19fde2815e695d52, I find that it try to read /mnt/us/extensions/kpvbooklet/bin/booklet.ini and run /mnt/us/extensions/kpvbooklet/bin/booklet.sh.
However, no file exists there. I think maybe you should commit the booklet.sh and a default configuration of booklet.ini. Thanks a lot.

**Comment by p-hash** [comment]
I have the same issue with 0.6.5 from github releases on my pw1

**Comment by chrox** [comment]
@cytown The `bin` directory in `extentions` is ignored in git track as the .gitignore file has an entry of `bin`. I fixed this problem. And could you submit a patch of missing `/mnt/us/extensions/kpvbooklet/bin/booklet.ini` file? Thanks.

**Comment by CaesarW** [comment]
@cytown @chrox  I've already seen the latest commit of the missing .sh file. will there be a new release for these fixes?

**Comment by chrox** [comment]
Pinging @cytown for the still missing `/mnt/us/extensions/kpvbooklet/bin/booklet.ini` file.

**Comment by cytown** [comment]
Hi all, for the reason we all knew, I missed the recent gmail notifications, I will do that asap.

**Comment by cytown** [comment]
@chrox I have checked the codes, it's fine to missing booklet.ini, the script will auto create one and has default values.

**Comment by CaesarW** [comment]
so @cytown is there a new release?

**Comment by cytown** [comment]
@chrox please help to make a new release, if need help, let me know.

**Comment by cytown** [comment]
@CaesarW I found @chrox release 0.66 in #30 , please check there...

**Comment by CaesarW** [comment]
@cytown the version 0.6.6 released a bit earlier does not include bin folder therefore can't be executed by KUAL. so the version you posted on the forum is usable?

**Comment by CaesarW** [comment]
@cytown I've checked the forum and didn't find your post :-(. sorry for asking so much but can you please upload a latest kpvbooklet release on Github with bin folder? the old version 0.6.5 doesn't have the bin folder after installing. thank you very much!

**Comment by CaesarW** [comment]
@cytown I've checked the forum and didn't find your post :-(. sorry for asking so much but can you please upload a latest kpvbooklet release on Github with bin folder? the old version 0.6.5 doesn't have the bin folder after installing. thank you very much!

**Comment by cytown** [comment]
please check this:
https://github.com/koreader/kpvbooklet/issues/29#issuecomment-304015153

**Comment by CaesarW** [comment]
@cytown The link provided is found 404.

:-(...

**Comment by chrox** [comment]
Fixed with 90d07ce37730e96d23a1b67043277908b1bcebd4.
